### PR TITLE
Add rgba16float support for canvas context

### PR DESCRIPTION
**Pull request by jchen10**
This extends the readbackFromWebGPUCanvas and canvas_clear tests to cover rgba16float.

Issue: #915

<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by jchen10** [comment]
@kainino0x PTAL

**Comment by kainino0x** [comment]
> @kainino0x PTAL

Please add me as a reviewer so this goes to my review inbox :)

**Comment by kainino0x** [comment]
> Issue: #

Please fill this out. I believe it's issue #915, right?
https://github.com/orgs/gpuweb/projects/3

File: src/webgpu/web_platform/reftests/README.txt [hunk 1]
```diff
@@ -8,7 +8,6 @@ This tests things like:
 - The canvas renders with the correct transfer function.
 - The canvas blends and interpolates in the correct color encoding.
 
-TODO(#915): Test all possible output texture formats (currently only needs rgba16float).
 TODO(#916): Test all possible ways to write into those formats (currently only testing B2T copy).
```
Comment (kainino0x) [inline src/webgpu/web_platform/reftests/README.txt:11]: unrelated but I just noticed this is outdated
```suggestion
TODO(#916): canvas_complex: Test all ways to write into textures (currently only testing copy methods).
```

File: src/webgpu/web_platform/reftests/README.txt [hunk 2]
```diff
@@ -8,7 +8,6 @@ This tests things like:
 - The canvas renders with the correct transfer function.
 - The canvas blends and interpolates in the correct color encoding.
 
-TODO(#915): Test all possible output texture formats (currently only needs rgba16float).
```
Comment (kainino0x) [inline src/webgpu/web_platform/reftests/README.txt:11]: we are going to want to expand the canvas_complex tests as well (actually they need rgba8unorm tests too - I forgot before). Much less urgent, of course, but we should keep tracking a TODO here.
`TODO(#915): canvas_complex: test rgba8unorm and rgba16float`

File: src/webgpu/web_platform/reftests/README.txt (hunk 1 above)
Comment (jchen10) [reply src/webgpu/web_platform/reftests/README.txt]: Done!

File: src/webgpu/web_platform/reftests/README.txt (hunk 2 above)
Comment (jchen10) [reply src/webgpu/web_platform/reftests/README.txt:11]: Sounds fair.

**Comment by jchen10** [comment]
@kainino0x PTAL

**Review by kainino0x — APPROVED** [review]

**Comment by kainino0x** [comment]
Apologies for the delay; sidetracked by performance review stuff.

**Pull request merged**